radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/ActionParserService.java

```diff
@@ -227,6 +227,11 @@ public ActionParserService(Addressing addressing) {
 		}
 	}
 
+	private static Result<ECPublicKey> pubKeyOrSystem(JSONObject element) {
```

Comment:
I think it's better to return `Result<Optional>` here instead of `Result` with `null` inside.